Sources/PackageCollections/PackageCollections.swift

```diff
@@ -26,14 +26,21 @@ public struct PackageCollections: PackageCollectionsProtocol {
     private let storageContainer: (storage: Storage, owned: Bool)
     private let collectionProviders: [Model.CollectionSourceType: PackageCollectionProvider]
     let metadataProvider: PackageMetadataProvider
-    let searchProvider: PackageSearchProvider
 
     private var storage: Storage {
         self.storageContainer.storage
     }
 
     // initialize with defaults
     public init(configuration: Configuration = .init(), observabilityScope: ObservabilityScope) {
+        self.init(configuration: configuration, customMetadataProvider: nil, observabilityScope: observabilityScope)
+    }
+    
+    init(
+        configuration: Configuration = .init(),
+        customMetadataProvider: PackageMetadataProvider?,
+        observabilityScope: ObservabilityScope
+    ) {
```

Comment:
maybe collapse the two initializers into one by giving customMetadataProvider default value?

Reply:
I want to keep `customMetadataProvider` internal though